### PR TITLE
Minimal updates to bring old cheatsheet relatively up to date

### PR DESCRIPTION
**Pull request by parlough**
I'm not sure how valuable this article is with the codelab cheatsheet and the language tour, but I thought I could for now at least make some minor updates.

Does this article get traffic?

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Comment by kwalrath** [comment]
> I'm not sure how valuable this article is with the codelab cheatsheet and the language tour, but I thought I could for now at least make some minor updates.

Thank you! @filiph and I have discussed creating a real, printable one-pager (for I/O 2020, sigh), so I think a Dart cheatsheet could be very handy. It should be reformatted/rewritten and then made easier to find, but keeping it correct is a good first step.

> Does this article get traffic?

It gets some traffic. It is the 106th most visited page on dart.dev (out of ~150), just behind /tutorials/libraries/shared-pkgs and ahead of /web/debugging. 🤷‍♀️  In terms of average time-on-page, it's 38th, which I think indicates that people find it interesting.

Btw the highest time-on-page page, by far, is /codelabs/async-await.

File: src/_guides/language/cheatsheet.md [hunk 1]
```diff
@@ -33,9 +32,9 @@ Equivalent: `new Map<>();`
 
 ## Declaring fields
 
-| `var` | Generic `var` (dynamic) |
-| `final` | Same as `var` but final |
-| `const` | Constant |
+| `var` | Generic `var` with type inference |
+| `final` | Same as `var` but cannot be reassigned |
+| `const` | Compile time constant |
```
Comment (kwalrath) [inline src/_guides/language/cheatsheet.md:37]:
```suggestion
| `const` | Compile-time constant |
```

**Review by kwalrath — APPROVED** [review]
LGTM. Technically the compound modifier (compile-time) should be hyphenated because it's before the noun (https://developers.google.com/style/hyphens#compound-modifiers).

**Comment by parlough** [comment]
@kwalrath A printable cheatsheet sounds great and once I get a chance I'd also like to explore the concept more.

Thanks for the stats, that's interesting. I guess async/await can be pretty confusing.

**Pull request merged**